### ISSUE TITLE
connection refused when pretraining

### ISSUE THREAD

**Issue by gari-marcos** [xingdi-eric-yuan/GATA-public#3]
Hello, I am trying to pretrain action prediction and I get the following error:
```
Setting up a new session...
Exception in user code:
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connection.py", line 160, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/util/connection.py", line 80, in create_connection
    raise err
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/util/connection.py", line 70, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 603, in urlopen
    chunked=chunked)
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 355, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/http/client.py", line 1262, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/http/client.py", line 1308, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/http/client.py", line 1257, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/http/client.py", line 1036, in _send_output
    self.send(msg)
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/http/client.py", line 974, in send
    self.connect()
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connection.py", line 183, in connect
    conn = self._new_conn()
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connection.py", line 169, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fec435d3278>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gari/.local/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 641, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/gari/.local/lib/python3.6/site-packages/urllib3/util/retry.py", line 399, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8097): Max retries exceeded with url: /env/main (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fec435d3278>: Failed to establish a new connection: [Errno 111] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/site-packages/visdom/__init__.py", line 711, in _send
    data=json.dumps(msg),
  File "/home/gari/anaconda3/envs/gata/lib/python3.6/site-packages/visdom/__init__.py", line 677, in _handle_post
    r = self.session.post(url, data=data)
  File "/home/gari/.local/lib/python3.6/site-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/gari/.local/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/gari/.local/lib/python3.6/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/gari/.local/lib/python3.6/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8097): Max retries exceeded with url: /env/main (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fec435d3278>: Failed to establish a new connection: [Errno 111] Connection refused',))
[Errno 111] Connection refused
```
Could you tell me what does it mean and how to fix it?
Thank you in advance.

**Comment by MarcCote** [comment]
Hi @gari-marcos, by default we use Visdom to plot training curves, so it needs to be running. I made a PR #4  to disable it by default. I also updated the readme to explain how to enable it if desired (gist: you need to start `visdom` in a second terminal and open the link displayed by Vidsom in your browser.

**Comment by MarcCote** [comment]
@gari-marcos Can you try the PR mentioned above and let us know if that works for you?

**Comment by gari-marcos** [comment]
I am running ```visdom````and I get the following message: 
```
/home/gari/Documentos/TextWorld/git/GATA-public/evaluate.py:392: RuntimeWarning:

overflow encountered in exp
```
Moreover, my visdom windows is empty, I don't know if it should be like that.
Thank you so much.
PD: Sorry but I don't know how to pull a PR.

**Comment by xingdi-eric-yuan** [comment]
Should be fixed in PR [#4](https://github.com/xingdi-eric-yuan/GATA-public/pull/4), thanks for reporting!
